src/content/development/contribution-guide/_index.en.md

```diff
@@ -10,23 +10,38 @@ in order for their contributions to the project to be addressed in a timely mann
 
 ## Project Resources
 
-To be expanded
+Submariner uses [GitHub Projects] to manage releases. Read [Tracking Progress on your Project Board] to learn more on how to work with
+projects.
 
 ### Backlog Board
 
-To be expanded
+[The backlog board] hosts issues or features that are intended to be worked upon. While each issue is opened in
+its corresponding repository, this board gives an aggregated view of all open issues across all repositories. The board has total of seven
+columns. Any issue or epic opened should be assigned to `Backlog` Projects in `Backlog` column. Issues intenteded to be
+worked on for the next release are moved to `Next Version Candidate` column. During [pre-planning](#Pre-Planning), issues from the `Next
+Version Candidate` column are triaged. Issues or epics properly triaged, with assignees and priority labels addeded, are moved to [current
+release board](#Current Release Board). The next 3 columns are of priority. All unassigned issues are sorted into these three columns
+based on the priority discussed in the team meeting. Issues from these columns should be worked upon if the current release work is
+finished. Any leftover work from the previous release goes to the `Work in Progess column`. `Close?` column holds any issue/epic that
+could probably be closed for various reasons.
 
 ### Current Release Board
 
-To be expanded
+Current release work is tracked in [the 0.13 board]. THe board has 5 columns. The first one, `Schedule and Epics`, hosts all properly
+triaged epics targeted for the current release and schedule for the current release. Triaged issues are under `TO DO` column. When an
+issue is being worked on, it is moved to `In Progress` column. When a PR for the issue is pushed, the issue is moved to `In Review`
+column. The PR has the corresponding issue linked from the GitHub UI and is not tracked on the board. Once the PR is merged, the issue
```

Comment:
> corresponding issue linked from the GitHub UI and is not tracked on the board

I'm not really a fan of this practice because it reduces the visibility of PRs, which we very much need more visibility and review of.

Reply:
Agree. Also linking both issues and PRs to the board adds more noise. The right/preferred/recommended practise is to first open an issue, add it to the board and link the PR to the issue. The board shows the PRs linked to the issue inthe issue card.